src/tmpfiles/tmpfiles.c

```diff
@@ -1973,6 +2027,15 @@ static int parse_line(const char *fname, unsigned line, const char *buffer) {
         if (!should_include_path(i.path))
                 return 0;
 
+        r = specifier_expansion_from_arg(&i);
+        if (r == -ENOKEY) {
+                log_notice("[%s:%u] Uninitialized /etc detected, skipping rule", fname, line);
```

Comment:
In the paste you checked for machineid once up front. That chunk is missing from this patch. As @poettering said in the comment, there should be a log_warn for the first time %m is used without /etc/machine-id, and then only log_debug.

Reply:
> In the paste you checked for machineid once up front. That chunk is missing from this patch.

That check is now done implicitly in `specifier_machine_id_safe()`.

> there should be a log_warn for the first time %m is used without /etc/machine-id, and then only log_debug.

No log_warn, otherwise the patch becomes useless for the initial issue.

Here we log_notice each rule showing its location which seems better than a single and global notification. Maybe we could use log_debug...

Reply:
I don't think it's useful to get a warning (or notice) for every line. After all, if the %m lines fail, the solution is not to change them, but to initialize /etc/machine-id or change the time when tmpfiles is run. So one warning is completely sufficient.

Reply:
well I still prefer seeing a couple of notifications such as "specifier expansion not available, skipping rule in foo.conf, line 23" instead of a global "specifier expansion disabled due to ..." notification.